lib/experimental/class-wp-webfonts-provider-local.php

```diff
@@ -35,6 +35,33 @@ class WP_Webfonts_Provider_Local extends WP_Webfonts_Provider {
 	 */
 	protected $id = 'local';
 
+	/**
+	 * Holds a string which contains the type attribute for style tag.
+	 *
+	 * If the active theme does not declare HTML5 support for 'style',
+	 * then it initializes as `type='text/css'`.
+	 *
+	 * @since 6.1.0
+	 *
+	 * @var string
+	 */
+	private $type_attr = '';
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 6.1.0
+	 */
+	public function __construct() {
+		if (
+			function_exists( 'is_admin' ) && ! is_admin()
+			&&
+			function_exists( 'current_theme_supports' ) && ! current_theme_supports( 'html5', 'style' )
+		) {
+			$this->type_attr = " type='text/css'";
+		}
+	}
```

Comment:
This design comes from the `WP_Styles::__construct()`. I suspect it was initialized in the constructor as styles could be printed multiple times.

What about for a web font provider?

The `get_css()` and `print_styles()` are public methods, meaning either could be invoked within a plugin or theme for custom needs. Putting the logic that initializes the attribute into the constructor means the code only runs once rather than with each print.

What do you think @aristath?